File: Frontend/RscQuiz/src/RscQuiz/scripts/classes/quiz.ts
﻿import { Question } from './question'
import { Team } from './team'

export class Quiz {
    id: number = 0;
    name: string = "";
    place: string = "";
    dateTime: string = "";
    description: string = "";
    maxMembersPerTeam: number = 4;
    isOpen: boolean = false;
    questions: [Question];
    teams: Array<Team>;
}

File: IOS/RSCQuiz/Classes/Common/Models/Quiz.swift
//
//  Quiz.swift
//  RSCQuiz
//
//  Created by Nikola Majcen on 26/11/2016.
//  Copyright © 2016 Nikola Majcen. All rights reserved.
//

import Foundation
import ObjectMapper

class Quiz {
    
    var id: Int?
    var name: String?
    var place: String?
    var dateTime: String?
    var quizDescription: String?
    var isOpen: Bool?
    var teams: [Team]?
    
    required convenience init() {
        self.init()
    }
    
    required init?(map: Map) { }
}

extension Quiz: Mappable {
    
    func mapping(map: Map) {
        id <- map["id"]
        name <- map["name"]
        place <- map["place"]
        dateTime <- map["dateTime"]
        quizDescription <- map["description"]
        isOpen <- map["isOpen"]
        // teams <- map["teams"]
    }
}
